src/prngs/fortuna.c

```diff
@@ -268,14 +278,14 @@ unsigned long fortuna_read(unsigned char *out, unsigned long outlen, prng_state
    fortuna_update_iv(prng);
 
    if (rijndael_setup(prng->fortuna.K, 32, 0, &prng->fortuna.skey) != CRYPT_OK) {
-      LTC_MUTEX_UNLOCK(&prng->fortuna.prng_lock);
-      return 0;
+      tlen = 0;
```

Comment:
IMO `goto LBL_UNLOCK;` is not necessary here at it is basically the next line after the above mentioned `tlen = 0` line